inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -228,108 +229,99 @@ std::vector<cav_msgs::TrajectoryPlanPoint> InLaneCruisingPlugin::compose_traject
 
-    double max_x = curve_points.back().x();
-    double current_dist = 0;
-    double step_size = config_.curve_resample_step_size;
-    int current_speed_index = 0;
+  // compute subcurves to get correct time steps
+  std::vector<DiscreteCurve> sub_curves = compute_sub_curves(time_bound_points);
+  int parameter_length = 0;
+  double step_size = config_.curve_resample_step_size;
+  for (auto const discrete_curve : sub_curves)
+  {
+    double max_x = discrete_curve.points.back().point.x();
+    parameter_length += max_x / step_size;
+  }
 
-    while (current_dist < max_x - step_size) // Resample curve at tighter resolution
-    {
-      double x = current_dist;
-      double y = (*fit_curve)(x);
-      lanelet::BasicPoint2d p(x, y);
-      sampling_points.push_back(p);
+  int current_speed_index = 0;
+  int total_point_size = curve_points.size();
 
-      for (size_t i = current_speed_index; i < curve_points.size(); i++)
-      {
-        if (curve_points[i].x() >= current_dist)
-        {
-          current_speed_index = i;
-          break;
-        }
-      }
+  double next_speed_parameter_threshold = parameter_length / total_point_size;
+  double scaled_parameter = 0.0;
 
-      distributed_speed_limits.push_back(speed_limits[current_speed_index]); // Identify speed limits for resampled points
-      current_dist += step_size;
+  for (size_t parameter = 0; parameter < parameter_length; parameter++) // Resample curve at tighter resolution
+  {
+    Eigen::VectorXf v = (*fit_curve)(scaled_parameter);
+    lanelet::BasicPoint2d p(v.x(), v.y());
+    
+    sampling_points.push_back(p);
+    if (parameter > next_speed_parameter_threshold)
+    {
+      next_speed_parameter_threshold += (double)parameter_length / (double) total_point_size;
+      current_speed_index ++;
     }
+    distributed_speed_limits.push_back(speed_limits[current_speed_index]); // Identify speed limits for resampled points
+    scaled_parameter += 1.0/parameter_length; //adding parameter_step_size
+  }
 
-    log::printDebugPerLine(sampling_points, &log::basicPointToStream);
-
-    std::vector<double> yaw_values = carma_wm::geometry::compute_tangent_orientations(sampling_points);
-
-    std::vector<double> curvatures = carma_wm::geometry::local_circular_arc_curvatures(
-        sampling_points, config_.curvature_calc_lookahead_count);  
-
-    curvatures = smoothing::moving_average_filter(curvatures, config_.moving_average_window_size);
+  log::printDebugPerLine(sampling_points, &log::basicPointToStream);
 
-    log::printDoublesPerLineWithPrefix("curvatures[i]: ", curvatures);
+  std::vector<double> yaw_values = carma_wm::geometry::compute_tangent_orientations(sampling_points);
 
-    std::vector<double> ideal_speeds =
-        trajectory_utils::constrained_speeds_for_curvatures(curvatures, config_.lateral_accel_limit);
+  std::vector<double> curvatures = carma_wm::geometry::local_circular_arc_curvatures(
```

Comment:
This is the method that the other branch is pending changes on, just to make sure I follow?